framework/PostProcessors/ComparisonStatisticsModule.py

```diff
@@ -32,8 +32,230 @@
+    zN = 20
+    intervalZ = (highZ - lowZ)/zN
+    for i in range(zN):
+      z = lowZ + intervalZ*i
+      fZTable[0].append(z)
+      fZTable[1].append(fZ(z))
+    retDict["f_z_table"] = fZTable
+    sumFunctionDiff = mathUtils.simpson(fZ, lowZ, highZ, 1000)
+    firstMomentFunctionDiff = firstMomentSimpson(fZ, lowZ,highZ, 1000)
+    varianceFunctionDiff = mathUtils.simpson(lambda x:((x-firstMomentFunctionDiff)**2)*fZ(x),lowZ,highZ, 1000)
+    retDict['sum_function_diff'] = sumFunctionDiff
+    retDict['first_moment_function_diff'] = firstMomentFunctionDiff
+    retDict['variance_function_diff'] = varianceFunctionDiff
+  return retDict
+
+def __processData(data, methodInfo):
+  """
+    Method to process the computed data
+    @ In, data, np.array, the data to process
+    @ In, methodInfo, dict, the info about which processing method needs to be used
+    @ Out, ret, dict, the processed data including the counts of the bins
+  """
+  ret = {}
+  if hasattr(data,'tolist'):
+    sortedData = data.tolist()
+  else:
+    sortedData = list(data)
+  sortedData.sort()
+  low = sortedData[0]
+  high = sortedData[-1]
+  dataRange = high - low
+  ret['low'] = low
+  ret['high'] = high
+  if not 'binMethod' in methodInfo:
+    numBins = methodInfo.get("numBins", 10)
+  else:
+    binMethod = methodInfo['binMethod']
+    dataN = len(sortedData)
+    if binMethod == 'square-root':
+      numBins = int(math.ceil(math.sqrt(dataN)))
+    elif binMethod == 'sturges':
+      numBins = int(math.ceil(mathUtils.log2(dataN) + 1))
+    else:
+      self.raiseADebug("Unknown binMethod " + binMethod, 'ExceptedError')
+      numBins = 5
+  ret['numBins'] = numBins
+  kind = methodInfo.get("kind", "uniformBins")
+  if kind == "uniformBins":
+    bins = [low + x * dataRange / numBins for x in range(1, numBins)]
+    ret['minBinSize'] = dataRange / numBins
+  elif kind == "equalProbability":
+    stride = len(sortedData) // numBins
+    bins = [sortedData[x] for x in range(stride - 1, len(sortedData) - stride + 1, stride)]
+    if len(bins) > 1:
+      ret['minBinSize'] = min(map(lambda x, y: x - y, bins[1:], bins[:-1]))
+    else:
+      ret['minBinSize'] = dataRange
+  counts = mathUtils.countBins(sortedData, bins)
+  ret['bins'] = bins
+  ret['counts'] = counts
+  ret.update(mathUtils.calculateStats(sortedData))
+  skewness = ret["skewness"]
+  delta = math.sqrt((math.pi / 2.0) * (abs(skewness) ** (2.0 / 3.0)) /
+                    (abs(skewness) ** (2.0 / 3.0) + ((4.0 - math.pi) / 2.0) ** (2.0 / 3.0)))
+  delta = math.copysign(delta, skewness)
+  alpha = delta / math.sqrt(1.0 - delta ** 2)
+  variance = ret["sampleVariance"]
+  omega = variance / (1.0 - 2 * delta ** 2 / math.pi)
+  mean = ret['mean']
+  xi = mean - omega * delta * math.sqrt(2.0 / math.pi)
+  ret['alpha'] = alpha
+  ret['omega'] = omega
+  ret['xi'] = xi
+  return ret
+
+
+
+def _getPDFandCDFfromData(dataName, data, csv, methodInfo, interpolation,
+                         generateCSV):
```

Comment:
Is it possible to combine this method with ```_getPDFandCDFfromWeightedData```

Reply:
Yes, it would be possible, but I think one of the goals is to remove the Comparison Statistics PostProcessor, so keeping them separate lets Comparison Statistics PostProcesor be removed later.